crates/metaslang/cst/src/query/parser.rs

```diff
@@ -41,13 +44,38 @@ pub(super) fn parse_query<T: KindTypes>(input: &str) -> Result<ASTNode<T>, Query
         .parse(input)
         .finish()
         .map(|(_, query)| query)
-        .map_err(|e| QueryError {
-            message: e.to_string(),
-            row: 0,
-            column: 0,
+        .map_err(|e| {
+            let text_index = compute_row_and_column(e.errors[0].0, input);
+            QueryError {
+                message: e.to_string(),
+                row: text_index.line,
+                column: text_index.column,
+            }
         })
 }
 
+fn compute_row_and_column(target: &str, from: &str) -> TextIndex {
+    let target_offset = from.offset(target);
+    let mut text_index = TextIndex::ZERO;
+    let mut from_iter = from.chars();
+    let Some(mut c) = from_iter.next() else {
```

Comment:
We need both the current and the next character in order to use `TextIndex::advance`. We're not skipping the first character because it's used in the first iteration of the loop.

Reply:
`windows()` won't work because `.chars()` returns a `Chars` which is not a slice. There may be a better way to satisfy the type system, but what I found is that you first need to `.collect()` the characters and then you can use slice, which is inefficient.

But, there's also the problem that `windows()` will stop short at the one-but-last character (because it's returning pairs), so we would need to special case for that.

Alternatively, we could not use `TextIndex::advance` and compute the line/column using a method that doesn't require peeking the next character. From inspection of that method, it seems we would only incorrectly interpret line breaks with single `\r` characters?

WDTY?

Reply:
I see. Let's keep it aside for now. Thanks for the investigation.